src/topology/sheaves/skyscraper.lean

```diff
@@ -0,0 +1,314 @@
+  end,
+  map_comp' := λ U V W iVU iWV,
+  begin
+    by_cases hW : p₀ ∈ unop W,
+    { have hV : p₀ ∈ unop V := le_of_hom iWV.unop hW,
+      simp only [dif_pos hW, dif_pos hV, eq_to_hom_trans] },
+    { rw [dif_neg hW, eq_comp_eq_to_hom], apply terminal_is_terminal.hom_ext }
+  end }
+
+section
+
+variables {p₀}
+
+lemma skyscraper_presheaf_obj_of_mem {U : opens X} (h : p₀ ∈ U) :
+  (skyscraper_presheaf p₀ S).obj (op U) = S := if_pos h
+
+lemma skyscraper_presheaf_obj_of_not_mem {U : opens X} (h : p₀ ∉ U) :
+  (skyscraper_presheaf p₀ S).obj (op U) = terminal C := if_neg h
+
+end
+
+/--
+A skyscraper sheaf is a sheaf supported at a single point: if `p₀ ∈ X` is a specified
+point, then the skyscraper sheaf `𝓕` with value `A` is defined by `U ↦ A` if `p₀ ∈ U` and
+`U ↦ *` if `p₀ ∉ A` where `*` is some terminal object.
+-/
+def skyscraper_sheaf : sheaf C X :=
+⟨skyscraper_presheaf p₀ S, λ c U s hs x hx,
+  ⟨dite (p₀ ∈ U)
+    (λ h, x (hs p₀ h).some_spec.some (hs p₀ h).some_spec.some_spec.1 ≫
+        eq_to_hom ((skyscraper_presheaf_obj_of_mem S (hs p₀ h).some_spec.some_spec.2).trans
+          (skyscraper_presheaf_obj_of_mem S h).symm))
+    (λ h, terminal.from c ≫ (eq_to_hom (skyscraper_presheaf_obj_of_not_mem S h).symm)),
+    λ V inc h,
+    begin
+      by_cases hV : p₀ ∈ V,
+      { have hU : p₀ ∈ U := le_of_hom inc hV,
+        split_ifs,
+        generalize_proofs h₁ h₂,
+        dsimp,
+        split_ifs,
+        have := hx (hom_of_le inf_le_left) (hom_of_le inf_le_right) h₂.some_spec.1 h rfl,
+        dsimp at this,
+        split_ifs at this with H, swap,
+        { exact false.elim (H ⟨h₁.some_spec.some_spec.2, hV⟩) },
+        rw [eq_comp_eq_to_hom, category.assoc, eq_to_hom_trans] at this,
+        rw [this, category.assoc, eq_to_hom_trans, category.assoc, eq_to_hom_trans,
+          eq_to_hom_refl, category.comp_id] },
+      { dsimp,
+        split_ifs,
+        rw [←category.assoc, eq_comp_eq_to_hom],
+        exact terminal_is_terminal.hom_ext _ _, }
+    end,
+    λ y (hy : x.is_amalgamation y),
+    begin
+      split_ifs,
+      { generalize_proofs h₁ h₂,
+        have := hy h₂.some h₂.some_spec.1,
+        dsimp at this,
+        split_ifs at this with H, swap,
+        { exact false.elim (H h₂.some_spec.2), },
+        rw [eq_comp_eq_to_hom] at this,
+        rw [this, eq_comp_eq_to_hom, category.assoc, eq_to_hom_trans, eq_to_hom_refl,
+          category.comp_id], },
+      { rw [←eq_comp_eq_to_hom],
+        exact terminal_is_terminal.hom_ext _ _, }
+    end⟩⟩
+
+end
+
+section
+
+-- In this section, we calculate the stalks for skyscraper presheaves.
+-- We need to restrict universe level.
+
+open topological_space
+open category_theory category_theory.limits
+open Top
+open opposite
+
+universes u v
+
+variables {X : Top.{u}} (p₀ : X) {C : Type v} [category.{u} C] (S : C) [has_terminal C]
+
+lemma mem_nhds_of_not_mem_closure_singleton {y : X} (h : ¬p₀ ⤳ y) :
+  ∃ (U : open_nhds y), p₀ ∉ U.1 :=
```

Comment:
This lemma isn't really useful outside this context, it is just a rearrangement of `(not_iff_not.mpr specializes_iff_forall_open).mp h`, so instead of moving to a meaningful location, I just marked this lemma as private